packages/eas-cli/src/utils/profiles.ts

```diff
@@ -47,16 +54,49 @@ export async function getProfilesAsync<T extends ProfileType>({
   return await Promise.all(results);
 }
 
+async function setNodeVersionFromFileAsync(
+  projectDir: string,
+  profile: BuildProfile<Platform>
+): Promise<void> {
+  if (profile?.node) {
+    return;
+  }
+  const nodeVersion = await getNodeVersionFromFileAsync(projectDir);
+  if (nodeVersion) {
+    Log.log(
+      `The EAS build profile does not specify a Node.js version. Using the version specified in .nvmrc: ${nodeVersion} `
+    );
+
+    profile.node = nodeVersion;
+  }
+}
+
+async function getNodeVersionFromFileAsync(projectDir: string): Promise<string | undefined> {
+  const nvmrcPath = path.join(projectDir, '.nvmrc');
+  // eslint-disable-next-line node/no-sync
+  if (!fs.existsSync(nvmrcPath)) {
+    return;
+  }
+
+  const nodeVersion = (await fs.readFile(nvmrcPath, 'utf8')).toString().trim();
+  if (!semver.valid(nodeVersion)) {
+    throw new Error(`Invalid node version in .nvmrc: ${nodeVersion}`);
+  }
+  return nodeVersion;
+}
+
 async function readProfileAsync<T extends ProfileType>({
```

Comment:
```suggestion
async function readProfileWithOverridesAsync<T extends ProfileType>({
```

I think this name better describes what this function is actually doing